src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt

```diff
@@ -517,12 +517,13 @@ namespace Akka.Actor
         public static Akka.Actor.DeployableDecider From(Akka.Actor.Directive defaultDirective, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<System.Type, Akka.Actor.Directive>> pairs) { }
         public static Akka.Actor.LocalOnlyDecider From(System.Func<System.Exception, Akka.Actor.Directive> localOnlyDecider) { }
     }
-    public class DedicatedThreadScheduler : Akka.Actor.SchedulerBase, Akka.Actor.IDateTimeOffsetNowTimeProvider, Akka.Actor.ITimeProvider
+    public class DedicatedThreadScheduler : Akka.Actor.SchedulerBase, Akka.Actor.IDateTimeOffsetNowTimeProvider, Akka.Actor.ITimeProvider, System.IDisposable
     {
-        public DedicatedThreadScheduler(Akka.Actor.ActorSystem system) { }
+        public DedicatedThreadScheduler(Akka.Configuration.Config config, Akka.Event.ILoggingAdapter log) { }
```

Comment:
**Breaking** public API change: all `IScheduler` implementations take a `Config` and `ILoggingAdapter` object in their constructors instead of an `ActorSystem` reference. This is because the `IScheduler` is instantiated *inside* the `ActorSystem` CTOR and this creates a bit of a weird code cycle that was unresolvable in an earlier implementation of this. This is a safer implementation and will only affect users who are using their own custom `IScheduler`s, but all they have to do is update the CTOR.